tornado-examples/src/main/java/uk/ac/manchester/tornado/examples/compute/MatrixVectorRowMajor.java

```diff
@@ -0,0 +1,337 @@
+
+        // Allocate local memory for reduction
+        float[] localSum = context.allocateFloatLocalArray(localSize);
+
+        int rowOffset = rowId * n;
+
+        // Each thread calculates partial dot product
+        float partialSum = 0.0f;
+        for (int j = localId; j < n; j += localSize) {
+            int matrixIdx = rowOffset + j;
+            partialSum += w.get(matrixIdx) * x.get(j);
+        }
+
+        // Store partial sum in local memory
+        localSum[localId] = partialSum;
+        context.localBarrier();
+
+        // Parallel reduction within workgroup
+        for (int stride = localSize / 2; stride > 0; stride >>= 1) {
+            if (localId < stride) {
+                localSum[localId] += localSum[localId + stride];
+            }
+            context.localBarrier();
+        }
+
+        return localSum[0];
+    }
+
+    /**
+     * Runs the benchmark for different matrix sizes and reports results
+     */
+    public static void main(String[] args) {
+        System.out.println("Matrix-Vector Multiplication Benchmark");
+        System.out.println("======================================");
+
+        // Default parameters
+        int inputDim = 8192;   // Default input dimension (columns)
+        int outputDim = 2048; // Default output dimension (rows)
+
+        // Parse command line arguments if provided
+        if (args.length >= 3) {
+            try {
+                inputDim = Integer.parseInt(args[0]);
+                outputDim = Integer.parseInt(args[1]);
+                int localWorkGroupSize = Integer.parseInt(args[2]);
+            } catch (NumberFormatException e) {
+                System.err.println("Error parsing dimensions. Using defaults.");
+            }
+        }
+
+        System.out.println("Configuration:");
+        System.out.println("- Input dimension (columns): " + inputDim);
+        System.out.println("- Output dimension (rows): " + outputDim);
+        System.out.println("- Local work group size: " + LOCAL_WORK_GROUP_SIZE);
+        System.out.println("- Warmup iterations: " + WARM_UP_ITERATIONS);
+        System.out.println("- Benchmark iterations: " + BENCHMARK_ITERATIONS);
+        System.out.println();
+
+        // Create data arrays
+        FloatArray input = new FloatArray(inputDim);
+        FloatArray weights = new FloatArray(inputDim * outputDim);
+        FloatArray outputParallel = new FloatArray(outputDim);
+        FloatArray outputPureTornado = new FloatArray(outputDim);
+        FloatArray outputSeq = new FloatArray(outputDim);
+
+        // Initialize data
+        System.out.println("Initializing data...");
+        fillRandomData(input, -1.0f, 1.0f);
+        fillRandomData(weights, -0.1f, 0.1f);
+
+        // Arrays for timing measurements
+        ArrayList<Long> sequentialTimers = new ArrayList<>();
+        ArrayList<Long> parallelTimers = new ArrayList<>();
+        ArrayList<Long> pureTornadoTimers = new ArrayList<>();
```

Comment:
Can you please refactor the names of the lists to reflect the APIs? 

It is currently confusing since both Parallel API and Kernel API are used to express parallelism.